### DISCUSSION TITLE
How to retain screen animation option when using useNavigationBuilder in react navigation

### DISCUSSION THREAD

**Discussion by daviddamilola** [react-navigation/react-navigation#12065] [Q&A]
I am building a custom tab navigation with the `useNavigationBuilder`, the content part of the screen is rendered using `{descriptors[route.key].render()}`. This however does not respect the animation set in screen options for each route. Can this be achieved? setting the animation on the view does not look great as on the first render its not reflecting either.

<img width="798" alt="Screenshot 2024-07-16 at 06 36 56" src="https://github.com/user-attachments/assets/10398a36-4481-48d3-a472-e0e020ecbb7d">

I want the slide-in animation when moving from orders to order details. Screen option for order-details is defined as

```
 {
      name: 'order-details',
      component: Order,
      options: {
        headerLeft: WithPreviousRouteComponent,
        headerTitle: '',
        headerTransparent: true,
        headerStyle: false,
        headerShown: true,
        animation: 'slide_from_right',
        animationDuration: ANIMATION_DURATION,
      },
    },

```

**Comment by satya164** [comment]
Building a custom navigator means you take full control of the UI, including layout, transition animations etc. It's upto you to render the screens and animate them appropriately.
